### ISSUE TITLE
Multiple services VIP

### ISSUE THREAD

**Issue by fgazelot** [fortinet-ansible-dev/ansible-galaxy-fortimanager-collection#68]
Hello,

I dont know if it's a real issue or a features request.. But, i dont understand why we can't create multiple services in 'fmgr_firewall_vip' : 

````
 - name: Configure virtual IP for IPv4.
      fortinet.fortimanager.fmgr_firewall_vip:
        bypass_validation: False
        adom: root
        state: present
        forticloud_access_token: "{{ fmgr_token_info.json.access_token }}"
        firewall_vip:
          arp-reply: "enable" #<value in [disable, enable]>
          color: 14
          extintf: "any"
          comment: "ansible-comment"
          name: "ansible-test-vip"
          extip: "1.2.3.4"
          nat44: "enable"
          mappedip: "5.6.7.8"
          service:  "SSH"                <=====   Service is a string and not a list of strings
          protocol: "tcp"
          type: "load-balance"
````

In API, we can use a list of string : 

````
{
    "id": "be104448-7607-468e-92b6-166d4821bdca",
    "method": "update",
    "params": [
        {
            "url": "pm/config/adom/root/obj/firewall/vip/ansible-test-vip",
            "data": {
                "_if_no_default": 0,
                "portforward": 0,
                "arp-reply": 1,
                "src-filter": null,
                "status": 1,
                "extintf": [
                    "any"
                ],
                "extip": "1.2.3.4",
                "mappedip": "5.6.7.8",
                "ipv6-mappedip": "::",
                "srcintf-filter": null,
                "service": [                     <=== here
                    "HTTPS",
                    "SSH"
                ],
                .
                .
                .
                .
            }
        }
    ]
}
````

Feel free to say me if I missed something in "ansible-galaxy-fortimanager-collection".
Do you have any idea to create a vip with multiple services ? 

Thank you in advance. I look forward to your response
Kind regards,
fgazelot

**Comment by dux-fortinet** [comment]
Hi @fgazelot ,

Thanks for raising this issue. This is because the "service" parameter in the old version of FortiManager API is a string, but it is a list in the new version of FortiManager API. In order to make sure backward compatible, we set it as a string rather than a list.

Currently, you can bypass this restriction by setting 'bypass_validation: True'

For example

```
 - name: Configure virtual IP for IPv4.
      fortinet.fortimanager.fmgr_firewall_vip:
        bypass_validation: True
        adom: root
        state: present
        forticloud_access_token: "{{ fmgr_token_info.json.access_token }}"
        firewall_vip:
          arp-reply: "enable" #<value in [disable, enable]>
          color: 14
          extintf: "any"
          comment: "ansible-comment"
          name: "ansible-test-vip"
          extip: "1.2.3.4"
          nat44: "enable"
          mappedip: "5.6.7.8"
          service:  ["HTTPS", "SSH"]
          protocol: "tcp"
          type: "load-balance"
```

We will fix this problem in FortiManager 2.3.0. And you can set "service" as a list or string without setting 'bypass_validation: True'.

Thanks,
Xinwei

**Comment by dux-fortinet** [comment]
Hi fgazelot,

This BUG has been fixed in [FortiManager  Ansible Collection 2.3.0](https://galaxy.ansible.com/ui/repo/published/fortinet/fortimanager/)

The input argument 'service' can be a string or a list of sting without setting 'bypass_validation: True'.

Best regards,
Xinwei
